### ISSUE TITLE
Still delay when controlling MINIR4M

### ISSUE THREAD

**Issue by chenlarry84** [AlexxIT/SonoffLAN#1232]
As title, I recently added MINIR4M in my Homeassistant via SonoffLAN.
However, there was still delay when I control on Homeassistant.
I also block the internet function of my MINIR4M in my router but still don't work smoothly.

Is there a possibility that MINIR4M isn't supported yet?
(MINIR4M is the Matter version of MINIR4)

This my Debug page.

```
2023-09-07 21:09:18 [D] SysInfo: {'installation_type': 'Home Assistant OS', 'version': '2022.8.7', 'dev': False, 'hassio': True, 'virtualenv': False, 'python_version': '3.10.5', 'docker': True, 'arch': 'aarch64', 'timezone': 'UTC', 'os_name': 'Linux', 'os_version': '5.15.32-v8', 'user': 'root', 'supervisor': '2023.08.3', 'host_os': 'Home Assistant OS 8.5', 'docker_version': '20.10.14', 'chassis': 'embedded', 'sonoff_version': '3.5.2 (2ad1cd7)'}
2023-09-07 21:09:18 [D] 1 devices loaded from Cloud
2023-09-07 21:09:18 [D] 1001f61473 UIID 0138 | {'tzInfo': {'tzName': 'Asia/Taipei', 'utcOffset': 480}, 'version': 8, 'fwVersion': '1.0.1', 'rssi': -35, 'sledOnline': 'on', 'swMode': 2, 'swCtrlReverse': 'off', 'relaySeparation': 0, 'configure': [{'outlet': 0, 'enableDelay': 0, 'width': 7000, 'startup': 'off'}], 'pulses': [{'outlet': 0, 'pulse': 'off', 'switch': 'off', 'width': 500}], 'switches': [{'outlet': 0, 'switch': 'off'}, {'outlet': 1, 'switch': 'off'}, {'outlet': 2, 'switch': 'off'}, {'outlet': 3, 'switch': 'off'}], 'autoUpgrade': {'enable': False, 'range': ['02:00', '04:00']}, 'pairingMode': True}
2023-09-07 21:09:18 [D] AUTO mode start
2023-09-07 21:09:18 [D] 1001f61473 <= Local3 | 192.168.161.241:8081 | {'fwVersion': '1.0.1', 'rssi': -39, 'sledOnline': 'on', 'swMode': 2, 'swCtrlReverse': 'off', 'relaySeparation': 0, 'configure': [{'outlet': 0, 'enableDelay': 0, 'width': 7000, 'startup': 'off'}], 'pulses': [{'outlet': 0, 'pulse': 'off', 'switch': 'off', 'width': 500}], 'switches': [{'outlet': 0, 'switch': 'off'}], 'triggerType': 18, 'autoUpgrade': {'enable': False, 'range': ['02:00', '04:00']}} | 2
2023-09-07 21:09:18 [D] 1001f61473 <= Local3 | 192.168.161.241:8081 | {'fwVersion': '1.0.1', 'rssi': -39, 'sledOnline': 'on', 'swMode': 2, 'swCtrlReverse': 'off', 'relaySeparation': 0, 'configure': [{'outlet': 0, 'enableDelay': 0, 'width': 7000, 'startup': 'off'}], 'pulses': [{'outlet': 0, 'pulse': 'off', 'switch': 'off', 'width': 500}], 'switches': [{'outlet': 0, 'switch': 'off'}], 'triggerType': 18, 'autoUpgrade': {'enable': False, 'range': ['02:00', '04:00']}} | 2
2023-09-07 21:09:19 [D] CLOUD None => True
2023-09-07 21:09:19 [D] Add 4 entities
``` 

And this is my diagnostics

[config_entry-sonoff-de9b755471a118e08d5d436c8682c641.json.txt](https://github.com/AlexxIT/SonoffLAN/files/12549534/config_entry-sonoff-de9b755471a118e08d5d436c8682c641.json.txt)
[](url)

Thank you!

**Comment by jesusvallejo** [comment]
Why not use Matter HA integration?

**Comment by chenlarry84** [comment]
> Why not use Matter HA integration?

I upgraded my HA operating system for Matter integration feature but there was some issue for adding Matter device.
Furthermore, my HACS somehow went down with Error information of unknown cause after upgrading HA operating system.

**Comment by burgerga** [comment]
For me the device switches fine, but it the detached functionality I want to enable.

With the Matter integration you get 1 device with 1 entity, more complex settings (trigger and detached mode) are not exposed. With SonoffLan you can view that there is more, but they're all disabled: 

Matter:
![image](https://github.com/AlexxIT/SonoffLAN/assets/907362/9756a00b-7f75-4ff4-b14b-af7c1b4044dd)

SonoffLAN:
![image](https://github.com/AlexxIT/SonoffLAN/assets/907362/73b2d48e-0943-4f99-96bb-548d50e8fd17)

Here is my diagnostics: 
[sonoff-4e0854fab2e79bc1fb387d34e629506d-MINIR4M-ff24c9fc54738d461448e77112e339be.json.txt](https://github.com/AlexxIT/SonoffLAN/files/12726920/sonoff-4e0854fab2e79bc1fb387d34e629506d-MINIR4M-ff24c9fc54738d461448e77112e339be.json.txt)

Please let me know how I can help!

**Comment by AlexxIT** [comment]
If you can control device without internet - it works over LAN. If you can see some delays with control - it is device problem. Can't do anything about this

**Comment by burgerga** [comment]
Should I open a separate issue for the minir4m not having all functionality via SonoffLan (e.g. detached mode)?

**Comment by AlexxIT** [comment]
Why you don't have? You have it on your screens.

**Comment by burgerga** [comment]
Yes, but it's non-functional, very similar to the minir4: https://github.com/AlexxIT/SonoffLAN/issues/1184
If you set detached to True, which can only be done in the ewelink app for the minir4m, you do not get switch events in homeassistants. 
In other words, you cannot control/see the switch and relay separately

**Comment by AlexxIT** [comment]
It's disabled because you not enable it. It disabled by default.

**Comment by burgerga** [comment]
Not sure what you mean, it is enabled in the ewelink app:
![image](https://github.com/AlexxIT/SonoffLAN/assets/907362/e7096462-0d77-4de7-834a-af6c5a436bec)

But it's still grayed out in SonoffLAN, and when you click it, it says not available:
![image](https://github.com/AlexxIT/SonoffLAN/assets/907362/8662d873-1d41-4c84-80a8-b9ad3e8686f8)
Same story with the LED

**Comment by AlexxIT** [comment]
<img width="517" alt="image" src="https://github.com/AlexxIT/SonoffLAN/assets/511909/0a555528-923e-4543-b371-f04dcbb0ce4a">

**Comment by burgerga** [comment]
Aahhh, ok, my bad, thanks!
I will run some tests in the coming week, and I'll let you know

**Comment by burgerga** [comment]
So I tried again, and after enabling all entities, I can enable and disable "Detached mode" from HA but I still don't receive switch events, so it is the same issue as https://github.com/AlexxIT/SonoffLAN/issues/1184 except that you can't flash Tasmota on the Matter-enabled version to solve it :(
